### ISSUE TITLE
Upgrade to latest react (16.2) and fix console errors and update deprecated methods to match to best practices 

### ISSUE THREAD

**Issue by vic-topcoder** [appirio-tech/connect-app#1564]
We should update to latest react (16.2 as of Dec 27). There are still issues with our components throwing either warnings or errors. Use [React Dev tools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en) to view all warnings and better debug the code.

![screen shot 12](https://user-images.githubusercontent.com/8484061/34391065-899d0b56-eaf8-11e7-8104-62e2d4ccfbc4.png)
![screen shot 13](https://user-images.githubusercontent.com/8484061/34391088-9c2d49f2-eaf8-11e7-910d-24a189f36359.png)

**Comment by anuradevtc** [comment]
!

**Comment by anuradevtc** [comment]
!

**Comment by aretheregods** [comment]
I'm getting a PropTypes error coming from 58 files using "appirio-tech-react-components" which is, in turn, using React 15.3.1 and referencing PropTypes from said version of React. Once I update connect-app to the most recent version of React, this error prevents the app from rendering on my machine, of course. You can see the error here, [here](https://drive.google.com/file/d/1ozA4pyMSSmWqb4H_1-GxyV9AtFPmrMFe/view?usp=sharing). You can also see an incomplete list (The list is long.) of some of the components in which appirio-tech-react-components are used, [here](https://drive.google.com/file/d/1CyyeH-YUTpF9jhcMzk1hRgKJVaahntJy/view?usp=sharing). In order to get rid of this error, the mentioned module will have to be updated separately. What's the guidance on how to handle said case?
Also, not to get completely off track, but that module is also using coffeescript, which is the subject of another issue ( #1424 ), just so you know.

**Comment by aretheregods** [comment]
@gondzo The solving of this issue seems directly tied to the solving of issue #1424 due to the fact that both of these issues tie to the incompatibility of appirio-tech/react-components in its current form with React v16 and the continuing and deepening tech-debt present in that repo. The errors I'm getting on compiling connect-app are coming from usage of that module. It seems that in order to update _this_ issue, one would have to also fix issue #1424, which is untenable for me at this time given the complexity and depth of both issues. Noting these facts, I'm relinquishing my assignment and opening this issue back up for pickup.
All the best to whoever picks this up after me. And good luck.

**Comment by gondzo** [comment]
we have solved a lot of those errors, but couldn't do this one in parallel with other major upgrades like #1629

**Comment by vic-topcoder** [comment]
Latest React is 16.4.1 already... https://github.com/facebook/react/releases

**Comment by vikasrohit** [comment]
I think we should avoid this for the current release as we are already doing heavy lifting in this release and don't want to get confused which change caused the issue.
